### ISSUE TITLE
Getting unrelated eslint error 

### ISSUE THREAD

**Issue by psonawane-servicetitan** [eslint/eslint#10807]
**Tell us about your environment**

* **ESLint Version:** 8.2.6
* **Node Version:** v10.0.0
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint 

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
       "arrow-parens": ["error", "as-needed"],
        "jsx-a11y/click-events-have-key-events": "off",
        "jsx-a11y/no-static-element-interactions": "off",
        "no-param-reassign": "warn",
        "no-underscore-dangle": "off",
        "quote-props": ["error", "as-needed"],
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
I ran the following simple js program. But I got an error at line this.getA(); which is totally unrelated here. The error said "stateless functional components should not use this (react/no-this-in-sfc)"

```js
class Demo {
    getA = () => 'a';

    hasBug = () => {
        this.getA();

        return null;
    };
}
```

**What did you expect to happen?**
 No error expected

**What actually happened? Please include the actual, raw output from ESLint.**
 Got this error - stateless functional components should not use this (react/no-this-in-sfc) for line ```js this.getA() ```

**Comment by g-plane** [comment]
Hey, @psonawane-servicetitan could you please provide your correct ESLint version because ESLint hasn't released 8.x.x version so far.
Also, this problem may come from `eslint-plugin-react` instead of ESLint itself.

**Comment by kaicataldo** [comment]
It also looks like we're missing some of your configuration, as I don't see the `react/no-this-in-sfc` rule in the provided configuration above.

**Comment by psonawane-servicetitan** [comment]
The eslint version used is 4.19.1
There is no rule added for ```react/no-this-in-sfc``` explicitly

**Comment by kaicataldo** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
